libraries/lib-numeric-formats/NumericConverter.cpp

```diff
@@ -151,676 +27,45 @@ different formats.
 
 **********************************************************************/
 #include "NumericConverter.h"
-#include "SampleCount.h"
-#include "Beats.h"
+#include "NumericConverterFormats.h"
+#include "NumericConverterRegistry.h"
+
+#include "Project.h"
+
+#include "formatters/ParsedNumericConverterFormatter.h"
 
 #include <cmath>
 #include <wx/setup.h> // for wxUSE_* macros
 #include <wx/wx.h>
 
-//
-// ----------------------------------------------------------------------------
-// BuiltinFormatString Struct
-// ----------------------------------------------------------------------------
-//
-/** \brief struct to hold a formatting control string and its user facing name
- * Used in an array to hold the built-in time formats that are always available
- * to the user */
-struct BuiltinFormatString
-{
-   NumericFormatSymbol name;
-   NumericConverter::FormatStrings formatStrings;
-
-   friend inline bool operator ==
-      (const BuiltinFormatString &a, const BuiltinFormatString &b)
-         { return a.name == b.name; }
-};
-
-void NumericField::CreateDigitFormatStr()
-{
-   if (range > 1)
-      digits = (int)ceil(log10(range-1.0));
-   else
-      digits = 5; // hack: default
-   if (zeropad && range>1)
-      formatStr.Printf(wxT("%%0%dd"), digits); // ex. "%03d" if digits is 3
-   else {
-      formatStr.Printf(wxT("%%0%dd"), digits);
-   }
-}
-
-static const TranslatableString BuildBeatsFormat() {
-   double bpm = BeatsPerMinute.Read();
-   int uts = UpperTimeSignature.Read();
-   int lts = LowerTimeSignature.Read();
```

Comment:
After deleting these lines, these preferences used only to initialize a new project or to show the values in the preference dialog.

There are written not only by the dialog, but also in ProjectTimeSignature whenever a project's time signature is changed.

Is that the right behavior?  It is not like how ProjectRate behaves, so it seems inconsistent.

Reply:
I'm not sure what dialog you are talking about. I do agree that what is done is not quite ideal.

Do we need this Setting at all? Do we want to have preferences for the default values? I don't really know, but I don't think it matters a lot now.

Reusing the last project rate for new projects is as good approach as the other ones.

Reply:
Aha, I was misled by these remaining, unnecessary lines in GUIPrefs.cpp, when I looked for all uses of the settings:
```
   BeatsPerMinute.Invalidate();
   UpperTimeSignature.Invalidate();
   LowerTimeSignature.Invalidate();
```

So there is in fact no longer anything in the dialog then to make them available, unlike for the preferred project rate.

Reply:
Can also remove:
```
   ChoiceSetting LowerTimeSigSettings{ LowerTimeSignature,
      { ByColumns, mLTSChoices, mLTSCodes },
      mDefaultLTSIndex
   };
```